schemas/restrictions.js

```diff
@@ -1,6 +1,9 @@
 /*!
  * Copyright (c) 2021 Digital Bazaar, Inc. All rights reserved.
  */
+import {pattern} from 'iso8601-duration';
+const iso8601Regex =
+  pattern.toString().substring(1, pattern.toString().length - 1);
```

Comment:
Cool, I think this improves readability and comprehension:

```suggestion
// convert Regex to a string
const iso8601RegexString = pattern.toString();
// remove leading and trailing slashes
const iso8601Regex =
  iso8601RegexString.substring(1, iso8601RegexString.length - 1);
```

Reply:
Updated. 

I am questioning the convention behind these updates instead of just using the plain regex. I understand large regex strings can take away from easy to read code and proper formatting, but since this is a schema I don't think that applies as much. If I were to come to this file while debugging, the code comments would be useful, but I would still need to log out what the regex actually is so I can debug that if needed. Also, since we are importing this from a third party library, there is a (low) possibility that what is in `pattern` could be updated, introducing a bug.

What is the value of importing `pattern` and converting it to the proper format, instead of just using the regex itself?

Reply:
It's DRY principles mainly.  But also it is often the case when using well maintained libraries that we get a test suite for free.  Which is the case with this module: https://github.com/tolu/ISO8601-duration/tree/master/test

Your point about using libraries is certainly a valid concern.  Presently, we trust that if the maintainers of those packages makes a change that would break our usage of the library, they would release the change as a semver **BREAKING** change.  We would not *automatically* get breaking major releases in our packages and therefore breakage would be avoided.  This is why we at DB need to pay careful attention to how we release code because we have folk building things on our stuff as well.

Now, a package maintainer *could* accidentally release a breaking change as a patch etc.  The solution for this is using package-lock files and/or pinning to specific versions.  This means that if our application works with version 1.5.1 of a dependency it will ALWAYS continue to work with v1.5.1 if we pinned to that version.  But in the common case, we are willing to risk possible breakage for the convenience of not having to constantly be updating package files in order to receive patches and new features in our dependency tree.

Reply:
Understood, thanks for the explanation.